pkg/types/defaulting.go

```diff
@@ -0,0 +1,27 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package types
+
+import (
+	corev1 "k8s.io/api/core/v1"
+)
+
+func (v *Var) Defaulting() {
```

Comment:
please just put this in kustomization.go next to Var, or extract the Var class into a var.go file.

The whole concept of the "types" package is questionable but not sure where to put these structs yet that would be better.

Reply:
I extract the `Var` type into `var.go` file.